packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/pipeline-editor.tsx

```diff
@@ -71,7 +74,13 @@ function useAggregationCompleter(
   return completer.current;
 }
 
+// We track the editor changes only onBlur to avoid too many events.
+// On mount of component, we set the initial value of the editor in
+// initialPipelineText and once onBlur is triggered, we update it.
+// Note: Not storing this in component state to avoid re-render.
```

Comment:
You probably want to use `ref` here: it's exactly for keeping some state without causing a re-render. If you open two collection tabs this global value will get messed up because it's not isolated per component